src/plugins/formulas/formulas.js

```diff
@@ -203,36 +250,233 @@ export class Formulas extends BasePlugin {
       });
     }
 
+    const hfSetUpProperly = this.setupHF();
+
+    if (!hfSetUpProperly) {
+      return;
+    }
+
+    // HyperFormula events:
+    this.hyperformula.on('valuesUpdated', (...args) => this.onHFvaluesUpdated(...args));
+
+    this.applyHFSettings();
+
     super.enablePlugin();
   }
 
   /**
    * Disables the plugin functionality for this Handsontable instance.
    */
   disablePlugin() {
-    // TODO add tests for this line
-    this.hyperformula.destroy();
-
     super.disablePlugin();
   }
 
+  /**
+   * Triggered on `updateSettings`.
+   */
+  updatePlugin() {
+    this.#settings = this.hot.getSettings()[PLUGIN_KEY];
+
+    this.applyHFSettings();
+
+    if (isDefined(this.#settings.sheetName) && this.#settings.sheetName !== this.sheetName) {
+      this.switchSheet(this.#settings.sheetName);
+    }
+
+    super.updatePlugin();
+  }
+
   /**
    * Destroys the plugin instance.
    */
   destroy() {
+    if (!this.hyperformula) {
+      return;
+    }
```

Comment:
If there is no HF we will never call `super.destroy()` and therefore plugin will not be destroyed? Not calling `super` may break the plugin lifecycle in HOT

Reply:
true!